Framework/DataHandling/inc/MantidDataHandling/MaskDetectors.h

```diff
@@ -80,14 +80,28 @@ class DLLExport MaskDetectors : public API::Algorithm {
   void init() override;
   void exec() override;
   void execPeaks(DataObjects::PeaksWorkspace_sptr WS);
-  void fillIndexListFromSpectra(std::vector<size_t> &indexList,
-                                const std::vector<specnum_t> &spectraList,
-                                const API::MatrixWorkspace_sptr WS);
-  void appendToIndexListFromWS(std::vector<size_t> &indexList,
-                               const API::MatrixWorkspace_sptr maskedWorkspace);
+  void
+  fillIndexListFromSpectra(std::vector<size_t> &indexList,
+                           const std::vector<specnum_t> &spectraList,
+                           const API::MatrixWorkspace_sptr WS,
+                           const std::tuple<size_t, size_t, bool> &range_info);
```

Comment:
structure needs definition and tuple is the part of the language. So question of the taste and changes are too big for just cosmetic.